snapcraft/internal/deltas/_deltas.py

```diff
@@ -0,0 +1,210 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2016 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+import logging
+import os
+import subprocess
+
+from ._errors import (
```

Comment:
This is a good idea but doesn't follow the style we have been going with.

To follow the style it would look more like

snapcraft.deltas.errors:
```python
class DeltaGenerationFailed(...):
   ...


class DeltaFormatIsNoneError(...):
   ...
...
...
```

In `snapcraft/deltas/__init__.py`:
```python
from . import errors
```

And in code that access this from outside the package:

```python
from snapcraft.internal.deltas.errors import DeltaGenerationFailed
...
except DeltaGenerationFailed:
```

Or:
```python
from snapcraft.internal import deltas

except deltas.errors.DeltaGenerationFailed:
```

Also, why `DeltaGenerationFailed` instead of `DeltaGenerationError`?